### DISCUSSION TITLE
Where to start code/UI wanting to save each apps' volume level?

### DISCUSSION THREAD

**Discussion by caesarsamsi** [kyleneideck/BackgroundMusic#501] [Q&A]
I would love to contribute the tiny use case I have in mind below. I cloned the repository, haven't built yet, but thinking ahead:

1. How do I edit the UI? Xcode, or other tool?
2. Where is the code that handles volume level changes?

I'm eager to explore!

Thanks! Caesar.

Use Case:

1. Most of the time I have a handful of apps running. They make sound of variable volumes. 
2. I use BGM to set the volume of each app that I care about, any other app I did not set a volume level for I'd like to mute
3. Over time ... I quit and restart apps, logout/login, shutdown, startup, reboot, etc.

As Is:

1. I have to redo all the volume level of the apps I did before. The volume levels were not persisted.

To Be:

1. I can set volume levels for apps that I care for.
2. Volume levels are persisted at the time they were changed (no big dialog of "SAVE VOLUME LEVELS (YES/NO)?") 
3. Nice to have: a "mute" checkbox for each app, when the state is checked, the app is muted

**Comment by kyleneideck** [comment]
Thanks for asking.

>How do I edit the UI? Xcode, or other tool?

In Xcode. Open up `BGMApp > BGMApp > MainMenu.xib`. Sometimes some parts are hidden at first, so you have to click around in the sidebar on the left.

> Where is the code that handles volume level changes?

Have a look at https://github.com/kyleneideck/BackgroundMusic/issues/180#issuecomment-674364878. It should point you to that code and give you some ideas about how you could store the app volumes.

Let me know if you get stuck or have any other questions.
